test/framework/flux.go

```diff
@@ -474,6 +540,25 @@ func (e *ClusterE2ETest) providerConfig(clusterConfGitPath string) (*providerCon
 			e.ClusterConfig.Spec.WorkerNodeGroupConfigurations[0].MachineGroupRef.Name,
 			etcdName,
 			machineConfigs)
+	case v1alpha1.CloudStackDatacenterKind:
```

Comment:
All the extra code here can be avoided with the new Flux framework I mentioned above, where you just need to pass the ClusterFillers to `UpgradeWithGitOps` and control what fields to update there.

Reply:
Does this mean those test cases in flux.go for cloudstack can be removed? since the workload_clusters_test.go have the flux clodstack test case (upgrade and horizonal and scale combined)

Reply:
workload_clusters_test only covers workload cluster flux upgrade. we still need management cluster's flux upgrade test (with the new flux framework). And this `case` block can be removed.

Reply:
when you say `management cluster`, you mean the mangement cluster in a multiple cluster `MulticlusterE2ETest` ? if so, where is the simliar code for vsphere ( I assume it already used the new flux framework)

Reply:
1. I mean the management cluster test in the simple/NOT MulticlusterE2ETest case
2. upgrade I mean the upgrade command -- anything that's testable for upgrade
3. no. upgrade K8s version only applies to workload cluster

Reply:
Thanks for confirm my understand. 
Is the test case in file flux_test.go you are looking for? 
```
func TestCloudStackKubernetes121GitopsOptionsFlux(t *testing.T) {
	provider := framework.NewCloudStack(t, framework.WithRedhat120())
	test := framework.NewClusterE2ETest(
			t,
			provider,
			framework.WithFlux(),
			framework.WithClusterFiller(
				api.WithKubernetesVersion(v1alpha1.Kube120),
				api.WithControlPlaneCount(1),
				api.WithWorkerNodeCount(1),
				api.WithStackedEtcdTopology(),
			),
	)

	test.RunClusterFlowWithGitOps(
		framework.WithClusterUpgradeGit(
			api.WithWorkerNodeCount(3),
		),
		provider.WithProviderUpgradeGit(
			framework.UpdateRedhatTemplate121Var(),
		),
	)
}
```

Reply:
case block removed since not being used after using newer flux test methods.